Content.Shared/Xenoarchaeology/Artifact/XAE/XAERandomTeleportInvokerSystem.cs

```diff
@@ -0,0 +1,29 @@
+using Content.Shared.Popups;
+using Content.Shared.Xenoarchaeology.Artifact.XAE.Components;
+using Robust.Shared.Random;
+using Robust.Shared.Timing;
+
+namespace Content.Shared.Xenoarchaeology.Artifact.XAE;
+
+public sealed class XAERandomTeleportInvokerSystem : BaseXAESystem<XAERandomTeleportInvokerComponent>
+{
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    [Dependency] private readonly SharedTransformSystem _xform = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
+
+    /// <inheritdoc />
+    protected override void OnActivated(Entity<XAERandomTeleportInvokerComponent> ent, ref XenoArtifactNodeActivatedEvent args)
+    {
+        if (!_timing.IsFirstTimePredicted)
+            return;
+        // todo: teleport person who activated artifact with artifact itself
+        var component = ent.Comp;
```

Comment:
This TODO makes sense, otherwise this is just an annoying effect.

Maybe it should be the closest person, though, not the activator?

Reply:
nah, it makes artifact activation more 'special'. Activation is not just triggering effect - after u unlock node - every time u click on artifact or use it in hand - it will activate all nodes that are 'highest in depth' + 'have non depleted durability'. Thats a bit wierd mechanic Emo decided to use, but it is kinda interesting.